slick-testkit/src/test/scala/slick/test/lifted/MySQLDMLTest.scala

```diff
@@ -17,11 +17,9 @@ class MySQLDMLTest {
       def value = column[String]("value")
       def * = (id, value) <> ((V.apply _).tupled, V.unapply)
     }
-    assertTrue("generates insert ignore statements",
-      TableQuery[T].insertOrUpdate(1).statements.mkString.startsWith("insert ignore"))
-    assertTrue("there are no update action if insert primary key only",
-      TableQuery[T].insertOrUpdate(1).statements.mkString.endsWith(")"))
+    assertTrue("generates query with update action if insert primary key only",
+      TableQuery[T].insertOrUpdate(1).statements.mkString.endsWith("on duplicate key update `id`=VALUES(`id`)"))
     assertTrue("generates query with update action if insert with value",
-      TableQuery[T2].insertOrUpdate(V("test", "test-value")).statements.mkString.endsWith("on duplicate key update `value`=VALUES(`value`)"))
+      TableQuery[T2].insertOrUpdate(V("test", "test-value")).statements.mkString.endsWith("on duplicate key update `id`=VALUES(`id`), `value`=VALUES(`value`)"))
```

Comment:
see also #1627.

Reply:
What does "see also" mean? Does this PR test and fix it?

Reply:
Yes, the problem of #1627 is table without soft syms(= primary key only table, that mean) causes wrong statement. This PR fix the problem.

Reply:
In #1879, for solving this problem, applied `insert ignore` statement was applied. But this solution caused other problems. (#2076, #2045) This PR solves them, too.